core/unix/os.c

```diff
@@ -7013,30 +7013,34 @@ pre_system_call(dcontext_t *dcontext)
 #ifdef LINUX
     case SYS_ppoll: {
         kernel_sigset_t *sigmask = (kernel_sigset_t *)sys_param(dcontext, 3);
-        size_t sizemask = (size_t)sys_param(dcontext, 4);
-        /* The original app's sigmask parameter is now NULL effectively making the syscall
-         * a non p* version, and the mask's semantics are emulated by DR instead.
-         */
-        dcontext->sys_param3 = (reg_t)sigmask;
-        set_syscall_param(dcontext, 3, (reg_t)NULL);
-        bool sig_pending = false;
-        if (!handle_pre_extended_syscall_sigmasks(dcontext, sigmask, sizemask,
-                                                  &sig_pending)) {
-            /* In old kernels with sizeof(kernel_sigset_t) != sizemask, we're forcing
-             * failure. We're already violating app transparency in other places in DR.
-             */
-            set_failure_return_val(dcontext, EINVAL);
-            DODEBUG({ dcontext->expect_last_syscall_to_fail = true; });
-            execute_syscall = false;
-        }
-        if (sig_pending) {
-            /* If there had been pending signals, we revert re-writing the app's
-             * parameter, but we leave the modified signal mask.
+        if (sigmask != NULL) {
```

Comment:
Use "if (sigmask == NULL) break;" to avoid indenting everything which will keep the code more readable.  Similarly for the other two cases.

Reply:
Done.